nemo/collections/nlp/models/nlp_model.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC
+
+import torch
+from megatron import mpu
+from pytorch_lightning.core.lightning import LightningModule
+from pytorch_lightning.overrides.data_parallel import LightningDistributedDataParallel
+
+from nemo.core.classes import ModelPT
+from nemo.utils import AppState, logging
+
+__all__ = ['NLPModel']
+
+
+class NLPModel(ModelPT, ABC):
+    def init_ddp_connection(self, global_rank: int, world_size: int, is_slurm_managing_tasks: bool = True) -> None:
```

Comment:
Why do we need "is_slurm_managing_tasks" parameter? 

If we do need it, let's set is_slurm_managing_tasks=None by default. 
Then do something like this:

```
if is_slurm_managing_tasks is None:
    if self._trainer is not None:
        is_slurm_managing_tasks = self._trainer.is_slurm_managing_tasks
    else:
       is_slurm_managing_tasks = (os.environ['SLURM_NTASKS'] is not None)
```

Something like this: https://github.com/PyTorchLightning/pytorch-lightning/blob/59d847254897ad07c61c1338e9177f906dbd35be/pytorch_lightning/trainer/connectors/slurm_connector.py#L27

Reply:
I'm having the "is_slurm_managing_tasks" parameter just so we can return the default PTL init_ddp_connection when not using Model Parallel.

Reply:
I don't know that we need to add any logic concerning that flag as it's a PTL function and we're calling that function as is. The only difference is, in the case of model parallel, we then have additional logic initializing the model parallel groups.